primitives/src/asset.rs

```diff
@@ -53,6 +56,24 @@ pub enum Asset<MI: MaxEncodedLen> {
     ParimutuelShare(MI, CategoryIndex),
 }
 
+/// The `MarketAsset` enum represents all types of assets available in the
+/// prediction market protocol.
+///
+/// # Types
+///
+/// * `MI`: Market Id
+#[cfg_attr(feature = "std", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+#[derive(Clone, Copy, Debug, Decode, Default, Eq, Encode, MaxEncodedLen, PartialEq, TypeInfo)]
+pub enum PredictionMarketAsset<MI: MaxEncodedLen> {
```

Comment:
```suggestion
pub enum PredictionMarketAsset<MI> {
```

Not sure if that's a good idea, but I don't think we need this requirement here.

Reply:
I guess it's for convenience?

Reply:
We will need that for `MaxEncodedLen` implementation for the `Pool` struct.

Reply:
Ok, I think we can get rid of this after merging https://github.com/zeitgeistpm/zeitgeist/pull/1197